lib/auth/auth_with_roles.go

```diff
@@ -3581,6 +3582,17 @@ func (a *ServerWithRoles) filterKubeServices(server types.Server) error {
 	// connect to them.
 	mfaParams := services.AccessMFAParams{Verified: true}
 
+	// Filter out agents that don't have support for moderated sessions access
+	// checking if the user has any roles that require it.
+	minSupportedModeratedSessions := semver.New(utils.VersionBeforeAlpha("9.0.0"))
+	agentVersion := semver.New(server.GetTeleportVersion())
+	if hasLocalUserRole(a.context.Checker) {
+		roles := a.context.Checker.(LocalUserRoleSet)
+		if roles.HasModeratedSessionsPolicies() && agentVersion.LessThan(*minSupportedModeratedSessions) {
```

Comment:
nit: can we move the `agentVersion` instantiation inside the next `if` block?
Otherwise, we might be computing it and not using its value

Reply:
We could also move `minSupportedModeratedSessions` to a package-level var, since it doesn't ever change.

Reply:
Done